### PR TITLE
fix: Pylint Import Check

### PR DESCRIPTION
**Pull request by CoshUS**
#### Why is this change necessary?
SAM CLI integration tests run with dev.txt requirements.
In the case of a dev library being imported into prod code, the tests will not raise an error. 

#### How does it address the issue?
By isolating Pylint into its own file, we can create an env with only prod libraries and pylint which enables the ability to check dev imports while linting.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 def read(*filenames, **kwargs):
     encoding = kwargs.get("encoding", "utf-8")
-    sep = kwargs.get("sep", os.linesep)
```
Comment (aahung) [inline setup.py:11]: I am not familiar with this file, when will this being executed>?

File: appveyor.yml [hunk 2]
```diff
@@ -61,26 +61,47 @@ for:
 
       # Install AWS CLI Globally outside of a venv.
       - "pip install awscli"
+
```
Comment (aahung) [inline appveyor.yml:64]: I think an explanation of what envs will be created, what is the motivation, what each env is for here is good for people without context.

**Review by aahung — COMMENTED** [review]
thanks for adding this check! small feedbacks and questions

File: setup.py (hunk 1 above)
Comment (CoshUS) [reply setup.py:11]: setup.py is used when pip install is called.
extras_require["pre-dev"] will be used with pip install -e ".[pre-dev]"

File: appveyor.yml (hunk 2 above)
Comment (CoshUS) [reply appveyor.yml:64]: Added more comments

File: appveyor.yml [hunk 3]
```diff
@@ -61,26 +61,48 @@ for:
 
       # Install AWS CLI Globally outside of a venv.
       - "pip install awscli"
+      
+      # Create pre-dev virtual env without installing dev depenencies
+      # This is used for import check for prod code
```
Comment (aahung) [inline appveyor.yml:66]:
```suggestion
      # This is used for import check for prod code to prevent importing packages from dev.txt
```

**Review by aahung — APPROVED** [review]
LGTM, thx!

File: appveyor.yml [hunk 4]
```diff
@@ -61,26 +61,48 @@ for:
...
       - "python -m virtualenv venv"
       - "venv\\Scripts\\activate"
       - "python --version"
+      - "deactivate"
 
     build_script:
-      # Activate virtualenv again on windows
+      # Install pre-dev dependencies
+      - "pre-dev-venv\\Scripts\\activate"
+      - "python -c \"import sys; print(sys.executable)\""
```
Comment (sriram-mv) [inline appveyor.yml:83]: nit: we could specifically say which env this is and what is the executable too for when we are debugging builds.

File: setup.py (hunk 1 above)
Comment (sriram-mv) [reply setup.py:11]: can we verify os.linesep on windows is \n too? I remember some time ago there was a \r\n that crept into it. I could be wrong.

File: setup.py [hunk 5]
```diff
@@ -18,7 +19,16 @@ def read(*filenames, **kwargs):
 
 def read_requirements(req="base.txt"):
     content = read(os.path.join("requirements", req))
-    return [line for line in content.split(os.linesep) if not line.strip().startswith("#")]
+    requirements = list()
+    for line in content.split("\n"):
```
Comment (sriram-mv) [inline setup.py:23]: same here should we look at os.linesep?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (CoshUS) [reply setup.py:11]: This was changed to "\n" specifically as `io.open` in `read(*filenames, **kwargs)` defaults to "\n" as universal line ending no matter on what system. It was technically broken without an issue previously and has been confirmed on Windows.

Comment (CoshUS) [reply setup.py:11]: https://docs.python.org/3/glossary.html#term-universal-newlines

File: appveyor.yml (hunk 4 above)
Comment (CoshUS) [reply appveyor.yml:84]: Added comment on its current use.

**Review by sriram-mv — APPROVED** [review]

**Pull request merged**